recipes/folly/all/conanfile.py

```diff
@@ -19,278 +21,263 @@ class FollyConan(ConanFile):
+            raise ConanInvalidConfiguration(f"{self.ref} requires these Boost components: {required_components}. Try with '-o boost/*:without_{required_components}=False'")
 
     def source(self):
-        files.get(self, **self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version], strip_root=False)
+
+    def _cppstd_flag_value(self, cppstd):
+        if is_msvc(self):
+            prefix = "c"
+            year = str(cppstd)
+            if year > "17":
+                year = "latest"
+        return f"{prefix}++{year}"
+
+    def generate(self):
+        env = VirtualBuildEnv(self)
+        env.generate()
+        tc = CMakeToolchain(self)
+        tc.variables["CMAKE_PROJECT_folly_INCLUDE"] = "conan_deps.cmake"
+        # Folly fails to check Gflags: https://github.com/conan-io/conan/issues/12012
+        tc.variables["CMAKE_TRY_COMPILE_CONFIGURATION"] = str(self.settings.build_type)
+
+        if is_apple_os(self) and cross_building(self):
+            # INFO: Folly fails to configure Mac M1 -> Mac Intel:
+            # CMake Error: try_run() invoked in cross-compiling mode, please set the following cache variables appropriately: HAVE_VSNPRINTF_ERRORS_EXITCODE (advanced)
+            for var in ["FOLLY_HAVE_UNALIGNED_ACCESS", "FOLLY_HAVE_LINUX_VDSO", "FOLLY_HAVE_WCHAR_SUPPORT", "HAVE_VSNPRINTF_ERRORS"]:
+                tc.cache_variables[f"{var}_EXITCODE"] = 0
+
+        # Folly is not respecting this from the helper https://github.com/conan-io/conan-center-index/pull/15726/files#r1097068754
+        tc.cache_variables["CMAKE_POSITION_INDEPENDENT_CODE"] = self.options.get_safe("fPIC", True)
+        # Relocatable shared lib on Macos
+        if is_apple_os(self):
+            tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0042"] = "NEW"
+        # Honor BUILD_SHARED_LIBS from conan_toolchain (see https://github.com/conan-io/conan/issues/11840)
+        tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0077"] = "NEW"
+        # Honor Boost_ROOT set by boost recipe
+        tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0074"] = "NEW"
+
+
+        # 2019.10.21.00 -> either MSVC_ flags or CXX_STD
+        if is_msvc(self):
+            cxx_std_value = self._cppstd_flag_value(self.settings.get_safe("compiler.cppstd", self._min_cppstd))
+            tc.cache_variables["MSVC_LANGUAGE_VERSION"] = cxx_std_value
+            tc.cache_variables["MSVC_ENABLE_ALL_WARNINGS"] = False
+            tc.cache_variables["MSVC_USE_STATIC_RUNTIME"] = is_msvc_static_runtime(self)
+            tc.preprocessor_definitions["NOMINMAX"] = ""
+
+        if not self.dependencies["boost"].options.header_only:
+            tc.cache_variables["BOOST_LINK_STATIC"] = not self.dependencies["boost"].options.shared
+
+        tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0074"] = "NEW"  # Honor Boost_ROOT set by boost recipe
```

Comment:
this is duplicated but not worth a recompilation

Reply:
Agreed! Thank you for spotting!